rir/src/compiler/pir/type.h

```diff
@@ -224,6 +227,8 @@ struct PirType {
         assert(isRType());
         PirType t = *this;
         t.flags_.set(TypeFlags::isScalar);
+        t.flags_.reset(TypeFlags::lazy);
```

Comment:
no, that's wrong. scalar does not imply eager! you could have a promise that is guaranteed to return a scalar.